# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/registry_policy.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/registry_policy.cc
--- a/security/sandbox/chromium/sandbox/win/src/registry_policy.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/registry_policy.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -2,12 +2,12 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "sandbox/win/src/registry_policy.h"
+
 #include <stdint.h>
 
 #include <string>
 
-#include "sandbox/win/src/registry_policy.h"
-
 #include "base/logging.h"
 #include "sandbox/win/src/ipc_tags.h"
 #include "sandbox/win/src/policy_engine_opcodes.h"
@@ -20,21 +20,20 @@
 
 static const uint32_t kAllowedRegFlags =
     KEY_QUERY_VALUE | KEY_ENUMERATE_SUB_KEYS | KEY_NOTIFY | KEY_READ |
-    GENERIC_READ | GENERIC_EXECUTE | READ_CONTROL | KEY_WOW64_64KEY |
-    KEY_WOW64_32KEY;
+    GENERIC_READ | GENERIC_EXECUTE | READ_CONTROL;
 
 // Opens the key referenced by |obj_attributes| with |access| and
 // checks what permission was given. Remove the WRITE flags and update
 // |access| with the new value.
 NTSTATUS TranslateMaximumAllowed(OBJECT_ATTRIBUTES* obj_attributes,
                                  DWORD* access) {
-  NtOpenKeyFunction NtOpenKey = NULL;
+  NtOpenKeyFunction NtOpenKey = nullptr;
   ResolveNTFunctionPtr("NtOpenKey", &NtOpenKey);
 
-  NtCloseFunction NtClose = NULL;
+  NtCloseFunction NtClose = nullptr;
   ResolveNTFunctionPtr("NtClose", &NtClose);
 
-  NtQueryObjectFunction NtQueryObject = NULL;
+  NtQueryObjectFunction NtQueryObject = nullptr;
   ResolveNTFunctionPtr("NtQueryObject", &NtQueryObject);
 
   // Open the key.
@@ -45,7 +44,7 @@
 
   OBJECT_BASIC_INFORMATION info = {0};
   status = NtQueryObject(handle, ObjectBasicInformation, &info, sizeof(info),
-                         NULL);
+                         nullptr);
   CHECK(NT_SUCCESS(NtClose(handle)));
   if (!NT_SUCCESS(status))
     return status;
@@ -63,7 +62,7 @@
                              ULONG* disposition,
                              HANDLE target_process) {
   *target_key_handle = nullptr;
-  NtCreateKeyFunction NtCreateKey = NULL;
+  NtCreateKeyFunction NtCreateKey = nullptr;
   ResolveNTFunctionPtr("NtCreateKey", &NtCreateKey);
 
   if (MAXIMUM_ALLOWED & desired_access) {
@@ -73,14 +72,14 @@
   }
 
   HANDLE local_handle = INVALID_HANDLE_VALUE;
-  NTSTATUS status = NtCreateKey(&local_handle, desired_access, obj_attributes,
-                                title_index, class_name, create_options,
-                                disposition);
+  NTSTATUS status =
+      NtCreateKey(&local_handle, desired_access, obj_attributes, title_index,
+                  class_name, create_options, disposition);
   if (!NT_SUCCESS(status))
     return status;
 
-  if (!::DuplicateHandle(::GetCurrentProcess(), local_handle,
-                         target_process, target_key_handle, 0, FALSE,
+  if (!::DuplicateHandle(::GetCurrentProcess(), local_handle, target_process,
+                         target_key_handle, 0, false,
                          DUPLICATE_CLOSE_SOURCE | DUPLICATE_SAME_ACCESS)) {
     return STATUS_ACCESS_DENIED;
   }
@@ -92,7 +91,7 @@
                            OBJECT_ATTRIBUTES* obj_attributes,
                            HANDLE target_process) {
   *target_key_handle = nullptr;
-  NtOpenKeyFunction NtOpenKey = NULL;
+  NtOpenKeyFunction NtOpenKey = nullptr;
   ResolveNTFunctionPtr("NtOpenKey", &NtOpenKey);
 
   if (MAXIMUM_ALLOWED & desired_access) {
@@ -107,15 +106,15 @@
   if (!NT_SUCCESS(status))
     return status;
 
-  if (!::DuplicateHandle(::GetCurrentProcess(), local_handle,
-                         target_process, target_key_handle, 0, FALSE,
+  if (!::DuplicateHandle(::GetCurrentProcess(), local_handle, target_process,
+                         target_key_handle, 0, false,
                          DUPLICATE_CLOSE_SOURCE | DUPLICATE_SAME_ACCESS)) {
     return STATUS_ACCESS_DENIED;
   }
   return STATUS_SUCCESS;
 }
 
-}
+}  // namespace
 
 namespace sandbox {
 
@@ -195,10 +194,10 @@
 
   UNICODE_STRING uni_name = {0};
   OBJECT_ATTRIBUTES obj_attributes = {0};
-  InitObjectAttribs(key, attributes, root_directory, &obj_attributes,
-                    &uni_name, NULL);
+  InitObjectAttribs(key, attributes, root_directory, &obj_attributes, &uni_name,
+                    nullptr);
   *nt_status = NtCreateKeyInTarget(handle, desired_access, &obj_attributes,
-                                   title_index, NULL, create_options,
+                                   title_index, nullptr, create_options,
                                    disposition, client_info.process);
   return true;
 }
@@ -220,10 +219,10 @@
 
   UNICODE_STRING uni_name = {0};
   OBJECT_ATTRIBUTES obj_attributes = {0};
-  InitObjectAttribs(key, attributes, root_directory, &obj_attributes,
-                    &uni_name, NULL);
+  InitObjectAttribs(key, attributes, root_directory, &obj_attributes, &uni_name,
+                    nullptr);
   *nt_status = NtOpenKeyInTarget(handle, desired_access, &obj_attributes,
-                                client_info.process);
+                                 client_info.process);
   return true;
 }
 